aidial_adapter_vertexai/chat/bison/prompt.py

```diff
@@ -17,31 +18,75 @@ def __repr__(self) -> str:
         return f"{self.value!r}"
 
 
-class BisonPrompt(BaseModel):
-    context: Optional[str]
-    messages: List[ChatMessage]
+class BisonPrompt(BaseModel, Truncatable):
+    context: Optional[str] = None
+    history: List[ChatMessage] = []
+    prompt: str
 
     @classmethod
     def parse(cls, messages: List[Message]) -> "BisonPrompt":
-        context, messages = _validate_messages_and_split(messages)
+        context, messages = _validate_and_split_messages(messages)
+        bison_messages = list(map(_parse_message, messages))
+
         return cls(
             context=context,
-            messages=list(map(_parse_message, messages)),
+            history=bison_messages[:-1],
+            prompt=bison_messages[-1].content,
         )
 
     @property
-    def user_prompt(self) -> str:
-        return self.messages[-1].content or ""
+    def has_system_message(self) -> bool:
+        return self.context is not None
 
-    @property
-    def history(self) -> List[ChatMessage]:
-        return self.messages[:-1]
+    def keep(self, index: int) -> bool:
```

Comment:
The naming is a bit misleading

This method does not define, what messages we keep, and what messages will be thrown away

It defines a set of **mandatory messages**,that we need to keep anyway

I think something like `mandatory_keep` is more informative for user

Reply:
Renamed to `is_required_message`